src/components/TrafficList/TrafficListComponent.tsx

```diff
@@ -0,0 +1,353 @@
+          <>{item.protocol}</>,
+          <>
+            {!!links.metrics && (
+              <Link key={`link_m_${item.icon}_${name}`} to={links.metrics} className={'virtualitem_definition_link'}>
+                View metrics
+              </Link>
+            )}
+          </>
+        ]
+      };
+    });
+  };
+
+  private getLinks = (item: TrafficListItem) => {
+    if (item.node.isInaccessible) {
+      return { detail: '', metrics: '' };
+    }
+
+    const detail = `/namespaces/${item.node.namespace}/${this.nodeTypeToType(item.node.type, true)}/${item.node.name}`;
+
+    const metricsDirection = this.props.direction === 'inbound' ? 'in_metrics' : 'out_metrics';
+    let metrics = `${history.location.pathname}?tab=${metricsDirection}`;
+
+    switch (item.node.type) {
+      case NodeType.APP:
+        // All metrics tabs can filter by remote app. No need to switch context.
+        const side = this.props.direction === 'inbound' ? 'source' : 'destination';
+        metrics += `& ${URLParam.BY_LABELS}= ${encodeURIComponent(side + '_app=' + item.node.name)}`;
+        break;
+      case NodeType.SERVICE:
+        if (item.node.isServiceEntry) {
+          // Service Entries should be only destination nodes. So, don't build a link if direction is inbound.
+          if (this.props.direction !== 'inbound' && item.node.destServices && item.node.destServices.length > 0) {
+            const svcHosts = item.node.destServices.map(item => item.name).join(',');
+            metrics += `&${URLParam.BY_LABELS}=${encodeURIComponent('destination_service_name=' + svcHosts)}`;
+          } else {
+            metrics = '';
+          }
+        } else {
+          // Filter by remote service only available in the Outbound Metrics tab. For inbound traffic,
+          // switch context to the service details page.
+          if (this.props.direction === 'outbound') {
+            metrics += `&${URLParam.BY_LABELS}=${encodeURIComponent('destination_service_name=' + item.node.name)}`;
+          } else {
+            // Services have only one metrics tab.
+            metrics = `${detail}?tab=metrics`;
+          }
+        }
+        break;
+      case NodeType.WORKLOAD:
+        // No filters available for workloads. Context switch is mandatory.
+
+        // Since this will switch context (i.e. will redirect the user to the workload details page),
+        // user is redirected to the "opposite" metrics. When looking at certain item, if traffic is *incoming*
+        // from a certain workload, that traffic is reflected in the *outbound* metrics of the workload (and vice-versa).
+        const inverseMetricsDirection = this.props.direction === 'inbound' ? 'out_metrics' : 'in_metrics';
+        metrics = `${detail}?tab=${inverseMetricsDirection}`;
+        break;
+      default:
+        metrics = '';
+    }
+
+    return { detail: detail, metrics: metrics };
+  };
+
+  private nodeTypeToType = (type: NodeType, isURL?: boolean): string => {
```

Comment:
This makes me feel that at some point we might need a proper inflector library to cover there kinds of string manipulation.

Reply:
i refuse to replace a simple function with a proper library of a type I have never even heard of :)